pallets/liquidity-pools-gateway/queue/src/lib.rs

```diff
@@ -207,7 +205,13 @@ pub mod pallet {
 			let mut processed_entries = Vec::new();
 
 			for (nonce, message) in MessageQueue::<T>::iter() {
-				processed_entries.push(nonce);
+				let remaining_weight = max_weight.saturating_sub(weight_used);
+				let next_weight = T::MessageProcessor::max_processing_weight(&message);
+
+				// We ensure we have still capacity in the block to process the message
+				if remaining_weight.all_gte(next_weight) {
+					break;
+				}
```

Comment:
We need to be sure we have capacity in the block before we start to compute the message. We can not add the weight later, it could be too late (even more risky with a batch with a lot of sub-messages)